eventuals/os.h

```diff
@@ -60,6 +64,161 @@ inline void SetAffinity(std::thread& thread, const size_t cpu) {
 }
 #endif // __MACH__
 
+#if defined(__linux__) || defined(__MACH__)
+
+////////////////////////////////////////////////////////////////////////
+
+namespace os {
+
+////////////////////////////////////////////////////////////////////////
+
+class Thread {
+ public:
+  Thread()
+    : thread_handle_{pthread_t{}},
+      joinable_(false) {}
+
+  ~Thread() noexcept {
+    CHECK(!joinable_) << "A thread was left not joined/not detached";
+  }
+
+  Thread(const Thread& other) = delete;
+
+  // IMPORTANT NOTE: on macos stacksize should be a multiple of the system
+  // page size!!!
+  // (check_line_length skip)
+  // https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/pthread_attr_setstacksize.3.html
+  template <typename Callable>
+  Thread(
+      Callable&& callable,
+      const std::string& name,
+      const Bytes& stack_size = Bytes(8'388'608))
```

Comment:
```suggestion
      const Bytes& stack_size = Megabytes(8))
```

Reply:
Nope! On macOS it seems that 8Mb is illegal size!
![1](https://user-images.githubusercontent.com/68085133/187452732-640fffb2-a619-471c-a8e9-3769bfc6129f.png)
So, I will keep `Bytes(8'388'608)`!

Reply:
????? But isn't Bytes(8'388'608) the same thing as Megabytes(8) ?????
I don't follow you, @ArthurBandaryk